src/status_im/contexts/wallet/create_account/new_keypair/check_your_backup/view.cljs

```diff
@@ -0,0 +1,121 @@
+(ns status-im.contexts.wallet.create-account.new-keypair.check-your-backup.view
+  (:require
+    [quo.core :as quo]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [status-im.contexts.wallet.common.temp :as temp]
+    [status-im.contexts.wallet.create-account.new-keypair.check-your-backup.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(def secret-words-count 12)
+
+(defn- random-selection
+  []
+  (->> (range secret-words-count)
+       (shuffle)
+       (take 4)
+       vec))
+
+(defn- random-words-with-string
+  [array given-string]
+  (let [random-words (->> array
+                          (remove #(= % given-string))
+                          (shuffle)
+                          (take 3))
+        result       (conj random-words given-string)]
+    (->> result
+         (shuffle)
+         vec)))
+
+(defn- cheat-warning
+  []
+  (let [{:keys [color]} (rf/sub [:wallet/current-viewing-account])]
+    [rn/view
+     [quo/drawer-top {:title (i18n/label :t/do-not-cheat)}]
+     [quo/text
+      {:style {:padding-horizontal 20
+               :padding-top        4
+               :padding-bottom     8}} (i18n/label :t/do-not-cheat-description)]
+     [quo/bottom-actions
+      {:button-one-label (i18n/label :t/see-recovery-phrase-again)
+       :button-one-props {:customization-color color
+                          :on-press            (fn []
+                                                 (rf/dispatch [:hide-bottom-sheet])
+                                                 (rf/dispatch [:navigate-back {:revealed? true}]))}}]]))
+
+(defn- button
+  [{:keys [word current-word quiz-index incorrect-count show-error? margin-right]}]
+  [quo/button
+   {:type            :grey
+    :on-press        (fn []
+                       (if (= word current-word)
+                         (do
+                           (reset! quiz-index (inc @quiz-index))
+                           (reset! incorrect-count 0)
+                           (reset! show-error? false))
+                         (do
+                           (when (> @incorrect-count 0)
+                             (rf/dispatch [:show-bottom-sheet
+                                           {:content cheat-warning}]))
+                           (reset! incorrect-count (inc @incorrect-count))
+                           (reset! show-error? true))))
+    :container-style (style/button margin-right)} word])
+
+(defn- buttons-row
+  [{:keys [margin-bottom options] :as params}]
+  [rn/view {:style (style/buttons-inner-container margin-bottom)}
+   [button
+    (merge params
+           {:word         (first options)
+            :margin-right 12})]
```

Comment:
we can `assoc` instead of merge

Reply:
@ulisesmac I don't think we can use `assoc` instead of `merge` here as there are multiple properties